### ISSUE TITLE
Faulty Consent String after CPC "DEACTIVATE ALL"

### ISSUE THREAD

**Issue by Waschnick** [as-ideas/oil#265]
For example
* on load, the consent strong is “BOk0p0DOk0p0DBQABBENCfAAAAApmAAA”, which doesn’t include any of the vendors ID’s,  which is correct. 
* If you “ACCEPT” then it becomes "BOk0p3_Ok0p3_BQABBENCf-AAAApmAbAAFAAoACwAGgAZAA4ACAAEgAKgAXAA0ACLAEwATQAoABSAC2AHEAPwAhABSACtAGUARYAnABOwCngGBAOoA” which includes all vendor ID’s, which is correct.
* However, if you got to “More Information”, and  "DEACTIVATE ALL”, or select some of the features on/off, you still get the consent string above, BOk0p3_Ok0p3_BQABBENCf-AAAApmAbAAFAAoACwAGgAZAA4ACAAEgAKgAXAA0ACLAEwATQAoABSAC2AHEAPwAhABSACtAGUARYAnABOwCngGBAOoA”. 
 
So basically, no matter how you customize it, once you hit “ACCEPT” you get the same result. We can’t find anything in configuration that could help with fixing that. 
 
Adding didn’t make a difference. Is there a way to 
* fix how “ACCEPT” buttons responds to the different settings? 
* add another button that opt our?  
* add a close button that opts out too?

**Comment by ltparis2018** [comment]
We analyzed the problem.
We could not reproduce the problem. Therefore, we guess there is a misunderstanding in respect of the expected behavior.
On load there is no consent. Therefore, the consent string (for example reported by the __cmp() function) does not include any vendor ID and any purpose.
If the user clicks ACCEPT this is an opt-in. It gives consent to all vendors and all purposes. Therefore, all of them are contained in the consent string.
If the user clicks “More information” and then uses the “Deactivate All” button this does not change anything until the user clicks the ACCEPT button. After that, the consent string does not contain any of the purposes anymore. But, all vendors are contained. Currently, OIL does not support opt-outs for vendors. This is still a planned feature – see GitHub issue https://github.com/as-ideas/oil/issues/205.
